web/frontend/src/components/camera.vue

```diff
@@ -34,108 +30,123 @@ const selectedMap = {
 } as const;
 
 const props = defineProps<Props>();
-const emit = defineEmits<Emits>();
 
-let pcdExpanded = $ref(false);
-let pointcloud = $ref<Uint8Array | undefined>();
-let camera = $ref(false);
+const videoEl = $ref<HTMLVideoElement>();
+const imgEl = $ref<HTMLImageElement>();
 
-const selectedValue = $ref('Live');
+let cameraOn = $ref(false);
+let cameraFrameIntervalId = $ref(-1);
+let camerasOn = $ref(0);
 
-const initStreamState = () => {
-  cameraStreamStates.set(props.cameraName, false);
-};
-
-const clearStreamContainer = (camName: string, elName: string) => {
-  const streamContainer = document.querySelector(
-    `[data-stream="${camName}"]`
-  );
-  if (elName === 'video') {
-    streamContainer?.querySelector('video')?.classList.add('hidden');
-    streamContainer?.querySelector('img')?.classList.remove('hidden');
-  } else {
-    streamContainer?.querySelector('img')?.classList.add('hidden');
-    streamContainer?.querySelector('video')?.classList.remove('hidden');
+const manageStreamStates = () => {
+  let counter = 0;
+  for (const value of cameraStreamStates.values()) {
+    if (value.name === props.cameraName && value.on) {
+      counter += 1;
+    }
   }
+  camerasOn = counter;
```

Comment:
We could maybe reduce the code here:

```suggestion
  camerasOn = cameraStreamStates.values().filter(({ name, on }) => name === props.cameraName && on).length;
```